### PR TITLE
Use logbook livestream when requesting a time window that includes the future

### PR DESCRIPTION
**Pull request by bdraco**
Requires https://github.com/home-assistant/core/pull/72258

## Proposed change

Use logbook livestream when requesting a time window that includes the future

Mobile

Main panel
May-21-2022 12-05-25
Areas
May-21-2022 12-09-16

More info
May-21-2022 12-07-38

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
I wish I could figure out how to make the new rows fade in

**Comment by bdraco** [comment]
Did some testing to make sure closing windows removes the subscriptions /developer-tools/event

File: src/panels/logbook/ha-logbook.ts [hunk 1]
```diff
@@ -253,21 +291,33 @@ export class HaLogbook extends LitElement {
       return;
     }
 
+    this._logbookEntries = [...newEntries].reverse();
+  }
+
+  private _appendNewEntries = (newEntries: LogbookEntry[]) => {
```
Comment (bdraco) [inline src/panels/logbook/ha-logbook.ts:297]: These two function should be smart enough to append or prepend the rows depending on the times.  An optimization we can do is if they select a long time window we can select the end of the time window right away and then deliver the older results in a second message

File: src/panels/logbook/ha-logbook.ts [hunk 2]
```diff
@@ -198,14 +187,30 @@ export class HaLogbook extends LitElement {
...
   }
 
+  private _unsubscribe(): void {
+    if (this._subscribed) {
+      this._subscribed.then((unsub) => unsub());
+      this._subscribed = undefined;
+    }
+  }
+
+  public disconnectedCallback() {
```
Comment (balloob) [inline src/panels/logbook/ha-logbook.ts:197]: if we add this, we should also add a `connectedCallback` or else we won't re-subscribe.

**Review by balloob — DISMISSED** [review]

File: src/panels/logbook/ha-logbook.ts (hunk 2 above)
Comment (bdraco) [reply src/panels/logbook/ha-logbook.ts:209]: 09351ac7cade90c2f90784d59cb01476ab22d148
f55d7650634430021dc168b932b12fd87b06c960

Comment (bdraco) [reply src/panels/logbook/ha-logbook.ts:209]: Need to go back to the history of the homeassistant websocket js api changes to remember how to test the disconnect/reconnect cycle

Comment (bdraco) [reply src/panels/logbook/ha-logbook.ts:209]: https://github.com/home-assistant/home-assistant-js-websocket/pull/255

👌

Will test after sleep

Comment (bdraco) [reply src/panels/logbook/ha-logbook.ts:209]: Added a testing disconnect button and hit it many times to make sure it reconnected ok 👍

File: src/panels/logbook/ha-logbook.ts (hunk 1 above)
Comment (bdraco) [reply src/panels/logbook/ha-logbook.ts]: efe9393d1709bd625004d236aa7479f00e9cad01

**Comment by bdraco** [comment]
I pulled the power on the access point my phone was connected to, watched it switch to 5g, continue to stream, and waited for the AP to power back up.  It successfully recovered streaming events during the switch over and switch back

**Comment by bdraco** [comment]
Still have one issue. It spins if you reload the page and the live stream is running.

**Comment by bdraco** [comment]
So a hard refresh `    let changed = changedProps.get("time");` will always be true twice

1. undefined
2. the time

**Comment by balloob** [comment]
that's weird it would be set to undefined. That is a bug.

**Review by balloob — APPROVED** [review]

**Review by zsarnett — APPROVED** [review]

**Pull request merged**